homeassistant/components/bayesian/binary_sensor.py

```diff
@@ -281,9 +305,9 @@ def _async_template_result_changed(event, updates):
                 self.hass, observations, text=f"{self._attr_name}/{entity}"
             )
 
-        all_template_observations = []
-        for value in self.observations_by_template.values():
-            all_template_observations.append(value[0])
+        all_template_observations: list[Observation] = []
```

Comment:
You can remove the conversion to bool on line 299:
```python
self._attr_is_on = bool(self.probability >= self._probability_threshold)
```
should be
```python
self._attr_is_on = self.probability >= self._probability_threshold
```

Reply:
done